pgptool-gui/src/main/java/org/pgptool/gui/tools/fileswatcher/MultipleFilesWatcher.java

```diff
@@ -36,230 +36,87 @@
 
 import org.apache.commons.io.FilenameUtils;
 import org.apache.log4j.Logger;
-import org.summerb.utils.threads.RecurringBackgroundTask;
 
-// NOTE: Consider migrating to org.apache.commons.io.monitor.FileAlterationObserver, it appears we introduced bycle here
+import org.apache.commons.io.monitor.FileAlterationListenerAdaptor;
+import org.apache.commons.io.monitor.FileAlterationMonitor;
+import org.apache.commons.io.monitor.FileAlterationObserver;
 
+/**
+ * @author Fredrick R. Brennan <copypaste@kittens.ph>
+ */
 public class MultipleFilesWatcher {
 	private static Logger log = Logger.getLogger(MultipleFilesWatcher.class);
 
 	private FilesWatcherHandler dirWatcherHandler;
 	private String watcherName;
-
-	private Map<WatchKey, BaseFolder> keys = new HashMap<>();
-	private Map<String, BaseFolder> baseFolders = new HashMap<>();
-
-	private Thread workerThread;
-	private WatchService watcher;
-
-	/**
-	 * since WatchService is not notifying us in case of parent folders removal AND
-	 * we don't want to overcomplicate code to watch ALL parent folders we're having
-	 * here another simple thread that will go and check folders existance
-	 */
-	private RecurringBackgroundTask dirsExistanceWatcherTask;
+	private FileAlterationMonitor monitor;
+	private Map<String, FileAlterationObserver> observers = new HashMap<>();
 
 	public MultipleFilesWatcher(FilesWatcherHandler dirWatcherHandler, String watcherName) {
 		this.dirWatcherHandler = dirWatcherHandler;
 		this.watcherName = watcherName;
+		this.monitor = new FileAlterationMonitor(1000);
```

Comment:
Please define constant for 1000 with self-descriptive name (including units of measurement)